src/profile-logic/import/chrome.js

```diff
@@ -0,0 +1,363 @@
+  IndexIntoFuncTable,
+  IndexIntoStackTable,
+} from '../../types/profile';
+
+import { getEmptyProfile } from '../../profile-logic/profile-data';
+import { getEmptyThread } from '../../test/fixtures/profiles/make-profile';
+import { ensureExists } from '../../utils/flow';
+
+// Chrome Tracing Event Spec:
+// https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+type TracingEventUnion = ProfileChunkEvent | ThreadNameEvent;
+
+type TracingEvent<Event> = {|
+  cat: string,
+  id: string,
+  ph: string, // Phase
+  pid: number, // Process ID
+  tid: number, // Thread ID
+  ts: number, // Timestamp
+  ...Event,
+|};
+
+type ProfileChunkEvent = TracingEvent<{|
+  name: 'ProfileChunk',
+  args: {
+    data: {
+      cpuProfile: {
+        nodes?: Array<{
+          callFrame: {
+            functionName: string,
+            scriptId: number,
+            lineNumber?: number,
+            columnNumber?: number,
+            url?: string,
+          },
+          id: number,
+          parent?: number,
+        }>,
+        samples: number[], // Index into cpuProfile nodes
+      },
+      timeDeltas: number[],
+    },
+  },
+  phase: 'P',
+|}>;
+
+type ThreadNameEvent = TracingEvent<{|
+  name: 'thread_name',
+  ph: 'm',
+  args: { name: string },
+|}>;
+
+type ScreenshotEvent = TracingEvent<{|
+  name: 'Screenshot',
+  ph: 'O',
+  args: { snapshot: string },
+|}>;
+
+export function isChromeProfile(profile: mixed): boolean {
+  if (!Array.isArray(profile)) {
+    return false;
+  }
+  const event = profile[0];
+  // Lightly check that some properties exist that are in the TracingEvent.
+  return (
+    typeof event === 'object' &&
+    event !== null &&
+    'ph' in event &&
+    'cat' in event &&
+    'args' in event
+  );
+}
+
+export function convertChromeProfile(profile: mixed): Promise<Profile> {
+  if (!Array.isArray(profile)) {
+    throw new Error(
+      'Expected an array when attempting to convert a Chrome profile.'
+    );
+  }
+  const eventsByName: Map<string, TracingEventUnion[]> = new Map();
+  for (const tracingEvent of profile) {
+    if (
+      typeof tracingEvent !== 'object' ||
+      tracingEvent === null ||
+      typeof tracingEvent.name !== 'string'
+    ) {
+      throw new Error(
+        'A tracing event in the chrome profile did not follow the expected form.'
+      );
+    }
+    const { name } = tracingEvent;
+    let list = eventsByName.get(name);
+    if (!list) {
+      list = [];
+      eventsByName.set(name, list);
+    }
+    list.push((tracingEvent: any));
+  }
+  return processTracingEvents(eventsByName);
+}
+
+type ThreadInfo = {
+  thread: Thread,
+  funcKeyToFuncId: Map<string, IndexIntoFuncTable>,
+  nodeIdToStackId: Map<number | void, IndexIntoStackTable | null>,
+  lastSeenTime: number,
+  lastSampledTime: number,
+};
+
+function getThreadInfo(
+  threadInfoByTid: Map<number, ThreadInfo>,
+  eventsByName: Map<string, TracingEventUnion[]>,
+  profile: Profile,
+  chunk
```

Comment:
Isn't this a `ProfileChunkEvent`? It looks like we can also give a type to this argument.

Reply:
Yup! Missed that one somehow. Thanks for catching this.